src/engine/math_util.c

```diff
@@ -333,16 +334,72 @@ void mtxf_identity(register Mat4 mtx) {
 }
 
 /// Set dest to a translation matrix of vector b.
-void mtxf_translate(Mat4 dest, Vec3f b) {
-    register s32 i;
-    register f32 *pen;
+UNUSED void mtxf_translate(Mat4 dest, Vec3f b) {
+    s32 i;
+    f32 *pen;
     for (pen = ((f32 *) dest + 1), i = 0; i < 12; pen++, i++) {
         *pen = 0;
     }
     for (pen = (f32 *) dest, i = 0; i < 4; pen += 5, i++) {
         *((u32 *) pen) = FLOAT_ONE;
     }
-    vec3f_copy(&dest[3][0], &b[0]);
+    vec3f_copy(dest[3], b);
+}
+
+/**
+ * Set matrix 'dest' to 'src' scaled by vector s
+ */
+void mtxf_scale_vec3f(Mat4 dest, Mat4 src, Vec3f s) {
+    f32 x = s[0];
+    f32 y = s[1];
+    f32 z = s[2];
+    f32 *dstp = (f32 *)dest;
+    f32 *srcp = (f32 *)src;
+    s32 i;
+    for (i = 0; i < 4; i++) {
+        dstp[ 0] = (srcp[ 0] * x);
+        dstp[ 4] = (srcp[ 4] * y);
+        dstp[ 8] = (srcp[ 8] * z);
+        dstp[12] =  srcp[12];
+        dstp++;
+        srcp++;
+    }
+}
+
+/**
+ * Multiply a vector with a transformation matrix, which applies the transformation
+ * to the point. Note that the bottom row is assumed to be [0, 0, 0, 1], which is
+ * true for transformation matrices if the translation has a w component of 1.
+ */
+UNUSED void mtxf_mul_vec3s(Mat4 mtx, Vec3s dest) {
+    f32 x = dest[0];
+    f32 y = dest[1];
+    f32 z = dest[2];
+    f32 *mtxp = (f32 *)mtx;
+    s16 *dstp = (s16 *)dest;
+    s32 i;
+    for (i = 0; i < 3; i++) {
+        *dstp++ = ((mtxp[ 0] * x)
+                 + (mtxp[ 4] * y)
+                 + (mtxp[ 8] * z)
+                 +  mtxp[12]);
+        mtxp++;
+    }
+}
```

Comment:
if this is unused, i dont think we should include it. this files pretty massive as is